amworkflow/occ_helpers.py

```diff
@@ -191,3 +206,245 @@ def topo_explorer(shape: TopoDS_Shape, shape_type: str) -> list:
+    @param vol TopoDS_Solid object representing a volume
+    @param gp_pnt If True return an gp_Pnt object otherwise a tuple of coordinates.
+    @return Center of mass of a volume.
+    """
+    props = GProp_GProps()
+    brepgprop_VolumeProperties(vol, props)
+    cog = props.CentreOfMass()
+    # Return the current coordinate of the current coordinate system.
+    if gp_pnt:
+        return cog
+    else:
+        return cog.Coord()
+    
+def get_face_area(face: TopoDS_Face) -> float:
+    """
+    @brief Get the area of a TopoDS_Face. This is an approximation of the area of the face.
+    @param face to get the area of.
+    @return The area of the face.
+    """
+    props = GProp_GProps()
+    brepgprop_SurfaceProperties(face, props)
+    face_area = props.Mass()
+    return face_area
+
+def get_faces(_shape):
+    """
+    @brief Get faces of a shape. This is a list of topods_Face objects that correspond to the faces of the shape
+    @param _shape shape to get faces of
+    @return list of topods_Face objects ( one for each face in the shape ) for each face
+    """
+    topExp = TopExp_Explorer()
+    topExp.Init(_shape, TopAbs_FACE)
+    _faces = []
+
+    # Add faces to the faces list
+    while topExp.More():
+        fc = topods_Face(topExp.Current())
+        _faces.append(fc)
+        topExp.Next()
+
+    return _faces
+
+def get_point_coord(p: gp_Pnt) -> tuple:
+    """
+    @brief Returns the coord of a point. This is useful for debugging and to get the coordinates of an object that is a part of a geometry.
+    @param p gp_Pnt to get the coord of
+    @return tuple of the coordinate of the point ( x y z ) or None if not a point ( in which case the coordinates are None
+    """
+    return p.Coord()
+
+def traverse(item: TopoDS_Shape) -> Topo:
+    return Topo(item)
+
+def create_face(wire: TopoDS_Wire) -> TopoDS_Face:
+    """
+    @brief Create a BRep face from a TopoDS_Wire. This is a convenience function to use : func : ` BRepBuilderAPI_MakeFace ` and
+    @param wire The wire to create a face from. Must be a TopoDS_Wire.
+    @return A Face object with the properties specified
+    """
+    return BRepBuilderAPI_MakeFace(wire).Face()
+
+def create_polygon(
+    points: list, isface: bool = True
+) -> TopoDS_Face or TopoDS_Wire:
+    """
+    @brief Creates a polygon in any shape. If isface is True the polygon is made face - oriented otherwise it is wires
+    @param points List of points defining the polygon
+    @param isface True if you want to create a face - oriented
+    @return A polygon
+    """
+    pb = BRepBuilderAPI_MakePolygon()
+    # Add points to the points.
+    for pt in points:
+        pb.Add(pt)
+    pb.Build()
+    pb.Close()
+    # Create a face or a wire.
+    if isface:
+        return create_face(pb.Wire())
+    else:
+        return pb.Wire()
+    
+def create_face_by_plane(pln: gp_Pln, *vt: gp_Pnt) -> TopoDS_Face:
```

Comment:
That's again just a wrapper. I would prefer not to add that and to use the make_face fct directly